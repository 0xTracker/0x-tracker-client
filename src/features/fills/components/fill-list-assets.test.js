import { render } from 'react-testing-library';
import React from 'react';

import FillListAssets from './fill-list-assets';

const erc20Asset = {
  amount: '150',
  tokenAddress: '0x12345',
  tokenSymbol: 'WETH',
  tokenType: 'Wrapped Ether',
  type: 'erc-20',
};

const erc721Asset = {
  amount: '1',
  tokenAddress: '0x12345',
  tokenId: 999,
  tokenSymbol: 'CK',
  tokenType: 'CryptoKitties',
  type: 'erc-721',
};

describe('fill list assets component', () => {
  it('should render for multiple assets', () => {
    const assets = [erc20Asset, erc721Asset];
    const { container } = render(<FillListAssets assets={assets} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render for ERC-20 asset', () => {
    const assets = [erc20Asset];
    const { container } = render(<FillListAssets assets={assets} />);

    expect(container).toMatchSnapshot();
  });

  it('should render for ERC-721 asset', () => {
    const assets = [erc721Asset];
    const { container } = render(<FillListAssets assets={assets} />);

    expect(container).toMatchSnapshot();
  });
});
