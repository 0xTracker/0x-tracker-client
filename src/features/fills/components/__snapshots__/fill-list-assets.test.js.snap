// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fill list assets component should link ERC-20 asset 1`] = `
<div>
  150
   
  <a
    class="link__InternalLink-sc-1m1n17q-1 IiDhz"
    href="/tokens/0x12345"
  >
    WETH
  </a>
</div>
`;

exports[`fill list assets component should link ERC-721 asset 1`] = `
<div>
  <a
    class="link__OutboundLink-sc-1m1n17q-0 iAzcTd"
    href="https://opensea.io/assets/0x12345/999"
    rel="noopener noreferrer"
    target="_blank"
  >
    CK
     #
    999
  </a>
</div>
`;

exports[`fill list assets component should render for ERC-20 asset 1`] = `
<div>
  150
   
  WETH
</div>
`;

exports[`fill list assets component should render for ERC-721 asset 1`] = `
<div>
  CK
   #
  999
</div>
`;

exports[`fill list assets component should render for multiple assets 1`] = `Multiple Assets`;
